src/components/SelectAssets/SelectAssetView.tsx

```diff
@@ -34,18 +33,26 @@ export const SelectAssetView = ({
         path={SelectAssetRoutes.Search}
         // This is already within a useMemo call, lint rule drunk
         // eslint-disable-next-line react-memo/require-usememo
-        element={<SelectAssets onBack={handleBack} onClick={onClick} />}
-      />
+      >
+        {<SelectAssets onBack={handleBack} onClick={onClick} />}
+      </Route>
     ),
     [handleBack, onClick],
   )
 
-  const redirectRoute = useMemo(() => <Route path='/' element={searchRedirect} />, [])
+  const redirectRoute = useMemo(
+    () => (
+      <Route path='/'>
+        <SelectAssets onBack={handleBack} onClick={onClick} />
+      </Route>
+    ),
+    [handleBack, onClick],
+  )
```

Comment:
The second guts of this - wrapping this into a dummy route fixes the issue with going back from address to select producing a flash

Reply:
(also on second read, I still have no idea why this works but it works)